### ISSUE TITLE
ui.table: Subscribing to 'update:pagination' breaks pagination

### ISSUE THREAD

**Issue by ghbm-itk** [zauberzeug/nicegui#2081]
### Description

I have a table object with pagination and I want to subscribe to the 'update:pagination' event to do stuff with the current page index. 

But when subscribing to the event the table gets stuck on page 1.

The event fires and I can get the new pagination information but the page doesn't change.

Minimal working example to reproduce:

```python
from nicegui import ui

table = ui.table(
    [
        {'id': 'Name', 'label': 'Name', 'field': 'Name', 'align': 'left'},
        {'id': 'Age', 'label': 'Age', 'field': 'Age', 'align': 'left'},
        {'id': 'Doodoo', 'label': 'Doodoo', 'field': 'Doodoo', 'align': 'left'}
    ],
    [
        {"Name": "John", "Age": "55", "Doodoo": "Dada"} for i in range(100)
    ],
    title="Pagination",
    pagination=5
)

table.on("update:pagination", lambda e: print(e.args))

ui.run()
```

**Comment by falkoschindler** [comment]
Hi @ghbm-itk,

when subscribing to such update events, Vue (or Quasar?) somehow doesn't apply the new value directly. Instead, the server needs to update the corresponding prop and send the update back to the client:
```py
def handle_pagination_change(e):
    print('pagination changed to', e.args)
    table._props['pagination'] = e.args
    table.update()

table.on('update:pagination', handle_pagination_change)
```

This is similar to what we do in `ui.tree`:

https://github.com/zauberzeug/nicegui/blob/a7bf8d2b6f4662a0e31c11599d5598210f8e3818/nicegui/elements/tree.py#L51-L69

At first I was sidetracked by a JavaScript error when starting the app, because the pagination update happens immediately when creating the table, before the websocket connection is created. But this issue doesn't prevent subsequent events from being emitted.

So I see multiple ways for improvement:

- Provide an official `on_pagination_change` event handler so that the user doesn't have to know these details.
- Add an explanation to the documentation about [generic events](https://nicegui.io/documentation/generic_events).
- Catch the JavaScript bug when emitting an event before establishing a websocket connection.

**Comment by ghbm-itk** [comment]
Hi thanks for the quick reply :)

On the topic of tables and events, I don't seem to be able to use the 'row-click' event on my table. It never fires.

Append my example above with:

```python
table.on('row-click', lambda e, r, i: ui.notify("Row click"))
```

**Comment by falkoschindler** [comment]
@ghbm-itk For some reason you need to pass the event name using camelCase. See https://nicegui.io/documentation/generic_events#specifying_event_attributes for a "rowClick" example.

I wonder if we should convert the name automatically, or if there could be any situation where it would break something... 🤔

**Comment by falkoschindler** [comment]
Ok, I made the following improvements:

- introduced an `on_pagination_change` event handler (d3b90486509906dbed10e97428399144dc9df6b5),
- support kebab-case event types (78d679c64d569a0207ef2654f68c1fe985d8ac7e)
- avoid JavaScript errors when the socket is not yet available (299d2f7a7e90e55e1964cb182fc4b5eebfa90c98)

This should be enough to close this issue.
